targets/TARGET_STM/TARGET_STM32G4/TARGET_STM32G431xB/system_clock.c

```diff
@@ -0,0 +1,186 @@
+/* mbed Microcontroller Library
+ * SPDX-License-Identifier: BSD-3-Clause
+ ******************************************************************************
+ *
+ * Copyright (c) 2019 STMicroelectronics.
+ * All rights reserved.
+ *
+ * This software component is licensed by ST under BSD 3-Clause license,
+ * the "License"; You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ ******************************************************************************
+ */
+
+/**
+  * This file configures the system clock as follows:
+  *-----------------------------------------------------------------
+  * System clock source | 1- USE_PLL_HSE_EXTC (external 8 MHz clock)
+  *                     | 2- USE_PLL_HSE_XTAL (external 8 MHz xtal)
+  *                     | 3- USE_PLL_HSI (internal 16 MHz)
+  *-----------------------------------------------------------------
+  * SYSCLK(MHz)         | 160 (default configuration) / 170 (CAN disabled)
+  * USB capable         | NO
```

Comment:
You should set as YES ?
(maybe with comment "to be set in mbed_app.json" ?)

Reply:
Oh, you removed the code for that.
So as you want, you can add back clock configuration for USB, or not.

Reply:
Ok, I'll add back the clock config then! Then it's just a targets.json edit to get the USB Device up and running.
Will modify to YES and add some comments. Need to test it a bit.

Looking at it, I think it actually needs to be in there for RNG as well. 
Otherwise the RNG clock will end up clocked from PLLQ (170 MHz).
I think you may want to edit it for G474RE as well? Something like this:

```
#if defined(DEVICE_TRNG) || defined(DEVICE_USBDEVICE)
    // Enable the HSI48 Clock (Can be used as a clocksource by CRS, RNG and USB)
    RCC_OscInitStruct.OscillatorType |= RCC_OSCILLATORTYPE_HSI48;
    RCC_OscInitStruct.HSI48State = RCC_HSI48_ON;
#endif
    if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK) {
        return 0; // FAIL
    }

#if defined(DEVICE_TRNG) || defined(DEVICE_USBDEVICE)
    // Connect the HSI48 Clock to drive the USB & RNG Clocks @ 48 MHz (CK48 Clock Mux)
    RCC_PeriphCLKInitStruct.PeriphClockSelection = RCC_PERIPHCLK_USB;
    RCC_PeriphCLKInitStruct.UsbClockSelection = RCC_USBCLKSOURCE_HSI48;
    if (HAL_RCCEx_PeriphCLKConfig(&RCC_PeriphCLKInitStruct) != HAL_OK) {
        return 0; // FAIL
    }
#endif
```

I'll checkin an updated version after verifying the test cases & usb connect.

Reply:
Ok, I ran the test suite! Same results as before. So should be good to go. Code is checked in.

Reply:
Actually, I see now that the RCC clock is connected to HSI48 in trng_api.c/trng_init using HAL_RCCEx_PeriphCLKConfig.

However, there is no similar construct in the init phase for USB (USBPhyHw.h/USBPhy_STM32.cpp).
I see that RCC_CCIPR/CLK48SEL resets to 00 (HSI48 Selected) though, so if it is preserved after reset USB should work without a HAL_RCCEx_PeriphCLKConfig statement in USBPhy_STM32.cpp (or in system_clock.c).

But is that good practice? Or better to keep the HAL_RCCEx_PeriphCLKConfig enabling statement in either system_clock.c or USBPhy_STM32.cpp?

I'll assume for now that no HAL_RCCEx_PeriphCLKConfig is needed in system_clock.c